File: design-patterns/ts/mediator/general/mediator.ts
// mediator.ts

import Colleague from "./colleague";

interface IMediator {
    sendMessage(colleague: Colleague, msg: string): void;
}

class Mediator implements IMediator {
    sendMessage(receiver: Colleague, msg: string): void {
        receiver.receiveMessage(msg);
    }
}

export default Mediator;

File: design-patterns/go/mediator/general/mediator.go
// mediator.go

package main

type Mediator struct {
}

func NewMediator() (mediator *Mediator) {
	mediator = &Mediator{}
	return
}

func (mediator *Mediator) SendMessage(receiver IColleague, msg string) {
	receiver.ReceiveMessage(msg)
}